### PR TITLE
Adding workflow to build the extension on push and pull request.

### PR DESCRIPTION
**Pull request by lahodaj**

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+      fail-fast: false
+    steps:
+
+      - name: Set up JDK ${{ matrix.java }}
+        uses: oracle-actions/setup-java@v1
+        with:
+          release: ${{ matrix.java }}
+
+      - name: Checkout ${{ github.ref }} ( ${{ github.sha }} )
+        uses: actions/checkout@v3
```
Comment (sormuras) [inline .github/workflows/main.yml:54]:
```suggestion
        uses: actions/checkout@v4
```

Use current version of the checkout action.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        with:
+          persist-credentials: false
+          submodules: false
+
+      - name: Checkout NetBeans
+        uses: actions/checkout@v3
```
Comment (sormuras) [inline .github/workflows/main.yml:60]:
```suggestion
        uses: actions/checkout@v4
```

Use current version of the checkout action here too.

**Review by sormuras — CHANGES_REQUESTED** [review]
Use current version, `@v4` as of today, of the checkout action.

**Comment by lahodaj** [comment]
> Use current version, `@v4` as of today, of the checkout action.

Thanks, applied!

**Review by sormuras — APPROVED** [review]
Workflow steps look good to me.

**Review by arvindaprameya — APPROVED** [review]

**Pull request merged**